src/opus/opus.cc

```diff
@@ -192,14 +201,43 @@ static bool update_replay_gain(OggOpusFile * opus_file,
     if (!track_gain)
         track_gain = album_gain;
 
-    /* Q7.8 number to float .. 2^8 = 256 */
-    /* EBU-R128 (-23 LUFS) to RG 2.0 (-18 LUFS) .. +5 dB */
-    rg_info->album_gain = str_to_double(album_gain) / 256 + 5.0f;
-    rg_info->track_gain = str_to_double(track_gain) / 256 + 5.0f;
+    if (got_r128)
+    {
+        /* Q7.8 number to float .. 2^8 = 256 */
+        /* EBU-R128 (-23 LUFS) to RG 2.0 (-18 LUFS) .. +5 dB */
+        rg_info->album_gain = str_to_double(album_gain) / 256 + 5.0f;
+        rg_info->track_gain = str_to_double(track_gain) / 256 + 5.0f;
+
+        /* Opus intentionally does not support peak tags */
+        rg_info->album_peak = 0;
+        rg_info->track_peak = 0;
+    }
+    else
+    {
+        rg_info->album_gain = str_to_double(album_gain);
+        rg_info->track_gain = str_to_double(track_gain);
 
-    /* Opus intentionally does not support peak tags */
-    rg_info->album_peak = 0;
-    rg_info->track_peak = 0;
+        const char * album_peak = opus_tags_query(tags, "REPLAYGAIN_ALBUM_PEAK", 0);
+        const char * track_peak = opus_tags_query(tags, "REPLAYGAIN_TRACK_PEAK", 0);
+
+        if (!album_peak && !track_peak)
+        {
+            /* okay, we have gain but no peak values */
+            rg_info->album_peak = 0;
+            rg_info->track_peak = 0;
+        }
+        else
+        {
+            /* fill in missing values if we can */
+            if (!album_peak)
+                album_peak = track_peak;
+            if (!track_peak)
+                track_peak = album_peak;
+
+            rg_info->album_peak = str_to_double(album_peak);
+            rg_info->track_peak = str_to_double(track_peak);
+        }
+    }
 
     AUDDBG("Album gain: %s (%f)\n", album_gain, rg_info->album_gain);
     AUDDBG("Track gain: %s (%f)\n", track_gain, rg_info->track_gain);
```

Comment:
Bring back the peak log messages as well. They were removed with https://github.com/audacious-media-player/audacious-plugins/commit/78f33f7316de481d6ce34ed3614076fda700aa1c.